tools/contrib/modify_singlept_site_neon.py

```diff
@@ -0,0 +1,435 @@
+#! /usr/bin/env python
+"""
+|------------------------------------------------------------------|
+|---------------------  Instructions  -----------------------------|
+|------------------------------------------------------------------|
+This script is for modifying surface dataset at neon sites
+using data available from the neon server.
+
+After creating a single point surface data file from a global 
+surface data file using subset_data.py, use this script to 
+overwrite some fields with site-specific data for neon sites.
+
+This script will do the following:
+- Download neon data for the specified site if it does not exist 
+    in the specified directory.
+- Modify surface dataset with downloaded data (neon-specific).
+
+-------------------------------------------------------------------
+Instructions for running on Cheyenne/Casper:
+
+load the following into your local environment
+    module load python
+    ncar_pylib
+
+To remove NPL from your environment on Cheyenne/Casper:
+    deactivate
+-------------------------------------------------------------------
+To see the available options:
+    ./modify_singlept_site_neon.py --help
+-------------------------------------------------------------------
+"""
+# TODO
+# if file not found run subset_data.py
+# Clean up imports for both codes...
+# Check against a list of valid names.
+
```

Comment:
I've added `zbedrock` functionality that we discussed in https://github.com/ESCOMP/CTSM/issues/1353#issue-865110750 here:

https://github.com/ESCOMP/CTSM/blob/0bf780e7b82ffbb13766882bb3dec193199d5ca1/tools/contrib/modify_singlept_site_neon.py#L512-L518

I used `zb_flag` for whether or not `zbedrock` is updated: so if `zbedrock` gets updated , the code updates the metadata to show `zbedrock` is modified:

https://github.com/ESCOMP/CTSM/blob/0bf780e7b82ffbb13766882bb3dec193199d5ca1/tools/contrib/modify_singlept_site_neon.py#L287-L291

Should we do anything special for LAI or Canopy height at this point? Maybe add it as a to-do item?

Reply:
@wwieder @negin513 has a question for you above. But since you normally want to run NEON with BGC I don't think we care about this for now, but I'll let Will weigh in for sure.

Reply:
we're always planning on running NEON with BGC for now.  if we ever get LAI data from NEON sites we can run in SP, but I don't see that happening soon.